### DISCUSSION TITLE
Symfony 6 Serializer normalize() return type declaration

### DISCUSSION THREAD

**Discussion by florianhuber81** [symfony/symfony#44775] [Q&A]
I use a custom normalizer to translate from DateTime() to UTCDateTime(), the MongoDB equivalent. Before Symfony 6 I had no problems. With the introduced retrun types I cannot return an object anymore. 

namespace Symfony\Component\Serializer;

Symfony 6
`public function normalize(mixed $data, string $format = null, array $context = []): array|string|int|float|bool|\ArrayObject|null` 

Before:
`public function normalize(mixed $data, string $format = null, array $context = []):`

Is there a way to change the return types in my implementation? Or any other way to return my desired UTCDateTime object? Or could it maybe be a feature request to add object to the retrun type list. 

regards
florian

**Comment by stof** [comment]
Adding `object` to that return type would also mean that we need to add it as an allowed *input* type of encoders. that would be quite a big BC break.

Note that the contract defined in phpdoc has not changed in Symfony 6. that union was already the expected return type in previous versions. The only difference is that the usage of the native return type detects when you don't respect the contract of the interface.

**Reply by florianhuber81** [reply]
Thanks for the quick replay. You are right, the phpdoc did not change. Any idea how I could attack that topic? To make it work for now I changed the Serializer.php file in the vendor folder 🤦‍♂️. I did not find any way to change the behaviour of the normalize() function.
